xdsl/transforms/scf_for_loop_range_folding.py

```diff
@@ -46,6 +46,9 @@ def match_and_rewrite(self, op: scf.For, rewriter: PatternRewriter) -> None:
                     return
                 folding_const = user.operands[0]
 
+            new_name_hint = (
+                user.result.name_hint if user.result.name_hint is not None else "new"
+            )
```

Comment:
This is cute but I worry that it might be a bit confusing if done extensively, like for the `new_lb_lb` above. I'm now tempted to say that we should not be doing anything here, and let the default name copying  done by the rewriter do the work.

Reply:
I agree, to be honest.

Reply:
Yep, I would just add the name hints to all variables in the test file, and remove the name hint manipulation in the rewrite